csrc/codegen.cpp

```diff
@@ -2964,10 +2964,13 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
       } else {
         indent() << "// Alias Allocation (changing dtype) - "
                  << alloc->memoryType() << "\n";
+        auto va = kernel_->summary().vectorized_accesses;
+        auto it = va.find(tv);
```

Comment:
We could instead use the original TV, whose vectorized accesses are guaranteed to be at least as large as this :man_shrugging:

Reply:
That may not work since the actual alignment size is `sizeof(buffer_dtype) * alias_alignment`. If we used the alignment of the original tensor, the alignment size would become larger if the size of the data type of the original tensor was smaller.

Reply:
This is currently only used by "inner aliasing" in which the size of the dtypes must match, as well as the parallelization and shapes. I'll leave it though for the reason you stated since in the future we might possibly allow mismatched shapes and dtype sizes subject to some other constraint.

Reply:
OK, I see. I think what's more important here would be we should make `CudaCodeGen` as trivial as possible and try to minimize the coupling with the lowering passes. It's difficult to keep track of all the assumptions made in the prior phases of code translations. For this case particular, since the lowering passes the alignment size, that should be just used as is. If some other valid value should be used instead, that decision should be made by the lowering and be recorded in the kernel summary.